common/thorhelper/thorsoapcall.cpp

```diff
@@ -1371,6 +1402,86 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
         }
     }
 
+    bool checkContentEncodingType(const char* encoding)
+    {
+        if (strieq(encoding, "gzip"))
+            return true;
+        return false;
+    }
+
+    bool checkContentDecoding(const StringBuffer& headers, StringBuffer& content, StringBuffer& contentEncoding)
+    {
+        if ((headers.length() == 0) || (content.length() == 0))
+            return false;
+
+        getHTTPHeader(headers.str(), CONTENT_ENCODING, contentEncoding);
+        if (contentEncoding.isEmpty())
+            return false;
+
+        if (!checkContentEncodingType(contentEncoding.str()))
+            throw MakeStringException(-1, "Content-Encoding:%s not supported", contentEncoding.str());
+        return true;
+    }
+
+    void decodeContent(const char* contentEncodingType, StringBuffer& content)
+    {
+        StringBuffer contentDecoded;
+        unsigned contentLength = content.length();
+        if (strieq(contentEncodingType, "gzip"))
```

Comment:
If you meant "checkConentEncodingSupported" then I understand why you didn't call it here.